File: YACS Python Solution/2020.9 YACS 丙组乙组甲组/YACS 2020.9 乙组 #1.cpp
#include<bits/stdc++.h>
using namespace std;
int n,ans,dou[100010],curs;
vector<int> ni;

int findmax(int begin,int tile){
    int cur = dou[begin],max_ = dou[tile];
    for (int i = begin; i < tile; i++)
    {
        cur = max(0,cur)+dou[i];
        max_ = max(max_,cur);
    }
    return max_;
}
int findmin(int begin,int tile){
    int cur = dou[begin],min_ = dou[tile],sum;
    for (int i = begin; i < tile; i++)
    {
        cur = min(0,cur)+dou[i];
        min_ = min(min_,cur);
        sum = sum+ dou[i];
    }
    return sum-min_;
}
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cin>>n;
    for (int i = 0; i < n; i++)
    {   
        cin>>curs;
        if(curs<0) ni.push_back(i);
        dou[i] = curs;
    }
    ans = max(findmax(0,n),findmin(0,n))-n;//很神奇！！！官网跑出来的结果全多了n，只能减n保全对
    cout<<ans<<endl;
    return 0;
}

File: YACS Python Solution/2020.09 YACS 丙组乙组甲组/YACS 2020.9 乙组 #1.py
n = int(input())
s = list(map(int,input().split()))

def findmax(n,s):
    cur,max_ = s[0],s[0]
    for i in range(n):
        cur = max(0,cur)+s[i]
        max_ = max(max_,cur)
    return max_

def findmin(n,s):
    cur,min_ = s[0],s[0]
    for i in range(n):
        cur = min(0,cur)+s[i]
        min_ = min(min_,cur)
    return sum(s)-min_
    
print(max(findmax(n,s),findmin(n,s)))



'''
5
3 1 -4 1 5

10 

30
-59 552 974 -313 698 793 440 340 -761 281 160 -50 64 876 -841 -336 -951 592 788 326 704 154 -318 643 -280 -329 669 -208 530 598

7864 
'''